### DISCUSSION TITLE
API check promotions available for customer before it gets added to the cart ?

### DISCUSSION THREAD

**Discussion by balanh-development** [swellstores/community#228] [Help] [answered]
So basically I want customers to see available promotions on their cart and need a way via the API to check for available promotions before it gets added to the cart ? Are swell-js or rest available for this ?

**Comment by ericingram** [comment, marked as answer]
@balanh-development when you pass `$promotions: true` to a cart or order, it will automatically calculate which promotions can be applied and add them to the record. You could use that functionality and remove them immediately if you want.

Something like this:

```
const cart = await swell.put('/carts/{id}', {
  id: ...,
  $promotions: true,
})

if (cart.promotions?.results) {
  // do something with promotions
  for (const promo of cart.promotions.results) {
    ...
  }
}

// remove promotions from the cart
const cart = await swell.put('/carts/{id}', { id: ..., promotion_ids: null })
```
